arch/arc/core/fault_s.S

```diff
@@ -295,6 +295,9 @@ exc_nest_handle:
 	bset.nz r3, r3, _ARC_V2_SEC_STAT_IRM_BIT
 	bclr.z r3, r3, _ARC_V2_SEC_STAT_IRM_BIT
 	sflag r3
+	/* save  _ARC_V2_SEC_STAT */
+	and r3, r3, 0xff
```

Comment:
Why do you clear the upper (reserved) bits here? in regular_irq.S below it isn't done - inconsistent.

Reply:
you mean "and r3, r3, 0xff", becuase the upper bits will be modified by exception, so need to clean it for fake interrupt.
In interrupt, the upper bits will not be modified , so no need to clear them.

I got your meaning,  it's more safe to do a clear in irq.  I will optmize it

Reply:
you mean "and r3, r3, 0xff", becuase the upper bits will be modified by exception, so need to clean it for fake interrupt.
In interrupt, the upper bits will not be modified , so no need to clear them.

I got your meaning, it's more safe to do a clear in irq. I will optmize it